diff --git a/src/webserver/x509.c b/src/webserver/x509.c
index 7520ff7..b8aa662 100644
--- a/src/webserver/x509.c
+++ b/src/webserver/x509.c
@@ -225,7 +225,7 @@ bool generate_certificate(const char* certfile, bool rsa, const char *domain, co
 	mbedtls_x509write_crt_set_basic_constraints(&ca_cert, 1, -1);
 
 	// Export CA in PEM format
-	if((ret = mbedtls_x509write_crt_pem(&ca_cert, ca_buffer, sizeof(ca_buffer))) != 0)
+	if((ret = MBEDTLS_X509WRITE_CRT_PEM(&ca_cert, ca_buffer, sizeof(ca_buffer))) != 0)
 	{
 		printf("ERROR: mbedtls_x509write_crt_pem (CA) returned %d\n", ret);
 		return false;
@@ -287,7 +287,7 @@ bool generate_certificate(const char* certfile, bool rsa, const char *domain, co
 		printf("mbedtls_x509write_crt_set_subject_alternative_name returned %d\n", ret);
 
 	// Export certificate in PEM format
-	if((ret = mbedtls_x509write_crt_pem(&server_cert, cert_buffer, sizeof(cert_buffer))) != 0)
+	if((ret = MBEDTLS_X509WRITE_CRT_PEM(&server_cert, cert_buffer, sizeof(cert_buffer))) != 0)
 	{
 		printf("ERROR: mbedtls_x509write_crt_pem returned %d\n", ret);
 		return false;
@@ -438,7 +438,7 @@ enum cert_check read_certificate(const char *certfile, const char *domain, const
 	mbedtls_pk_context key;
 	mbedtls_pk_init(&key);
 	bool has_key = true;
-	int rc = mbedtls_pk_parse_keyfile(&key, certfile, NULL);
+	int rc = MBEDTLS_PK_PARSE_KEYFILE(&key, certfile, NULL);
 	if (rc != 0)
 	{
 		log_info("No key found");
